packages/cli/src/fulltext/FileIndex.ts

```diff
@@ -298,3 +304,20 @@ export async function filterFiles(
   }
   return result;
 }
+
+// Reads git attribute patterns from a .gitattributes file.
+export function getGitAttributes(directory: string) {
+  const gitAttributesPath = join(directory, '.gitattributes');
+  if (!existsSync(gitAttributesPath)) return [];
+
+  const gitAttributesContent = readFileSync(gitAttributesPath, 'utf-8');
+  const lines = gitAttributesContent.split('\n').filter(Boolean);
+
+  return lines.map((line) => {
+    const [pattern, ...attributes] = line.split(/\s+/);
+    return {
+      pattern,
+      binary: attributes.includes('binary'),
+    };
+  });
+}
```

Comment:
Note this is much simpler than actual [gitattributes(5)](https://git-scm.com/docs/gitattributes):

> Each line in gitattributes file is of form:
> 
> pattern attr1 attr2 ...
> 
> That is, a pattern followed by an attributes list, separated by whitespaces. Leading and trailing whitespaces are ignored. Lines that begin with # are ignored. Patterns that begin with a double quote are quoted in C style. When the pattern matches the path in question, the attributes listed on the line are given to the path.

> When deciding what attributes are assigned to a path, Git consults $GIT_DIR/info/attributes file (which has the highest precedence), .gitattributes file in the same directory as the path in question, and its parent directories up to the toplevel of the work tree (the further the directory that contains .gitattributes is from the path in question, the lower its precedence). Finally global and system-wide files are considered (they have the lowest precedence).

This might be fine for our purposes, but the simplification warrants a comment at least — and I think skipping comments is easy enough to just do it.